collection.go

```diff
@@ -198,6 +198,17 @@ func NewCollectionWithOptions(spec *CollectionSpec, opts CollectionOptions) (col
 	}
 
 	for progName, origProgSpec := range spec.Programs {
+		shouldLoad := true
+		for _, exProg := range excludedPrograms {
+			if progName == exProg {
+				shouldLoad = false
+				break
+			}
+		}
+		if !shouldLoad {
+			continue
+		}
```

Comment:
It would be nice to clean this up by converting `excludedPrograms` into a set via `map[string]struct{}`. I don't feel very strongly about this though.